gprofiler/profilers/node.py

```diff
@@ -94,80 +106,77 @@ def _get_debugger_url() -> str:
     return cast(str, response_json[0]["webSocketDebuggerUrl"])
 
 
-def _send_socket_request(sock: WebSocket, cdp_request: Dict) -> None:
-    sock.send(json.dumps(cdp_request))
-    message = sock.recv()
-    try:
-        message = json.loads(message)
-    except json.JSONDecodeError:
-        raise NodeDebuggerUnexpectedResponse(message)
-
-    if (
-        "result" not in message.keys()
-        or "result" not in message["result"].keys()
-        or "type" not in message["result"]["result"].keys()
-        or message["result"]["result"]["type"] != "boolean"
-    ):
-        raise NodeDebuggerUnexpectedResponse(message)
-
-
-def _execute_js_command(sock: WebSocket, command: str) -> Any:
+@retry(NodeDebuggerProcessUndefined, 5, 0.5)
+def _evaluate_js_command(sock: WebSocket, command: str, expected_result: str) -> Any:
+    # Check if process or process.mainModule in command, and if it is, check if it is defined in js context
+    if "process.mainModule" in command:
+        command = f'typeof(process.mainModule) === "undefined" ? "process undefined" : {command}'
+    if "process" in command:
+        command = f'typeof(process) === "undefined" ? "process undefined" : {command}'
     cdp_request = {
         "id": 1,
         "method": "Runtime.evaluate",
         "params": {
             "expression": command,
-            "replMode": True,
         },
     }
     sock.send(json.dumps(cdp_request))
     message = sock.recv()
     try:
         message = json.loads(message)
     except json.JSONDecodeError:
-        raise NodeDebuggerUnexpectedResponse(message) from None
+        if expected_result == ResultType.NONE:
+            return message
+        else:
+            raise NodeDebuggerUnexpectedResponse(message) from None
     try:
-        return message["result"]["result"]["value"]
+        if (
+            isinstance(message["result"]["result"]["value"], str)
+            and message["result"]["result"]["value"] == "process undefined"
+        ):
+            raise NodeDebuggerProcessUndefined(message)
     except KeyError:
         raise NodeDebuggerUnexpectedResponse(message) from None
+    if (
+        "result" not in message.keys()
+        or "result" not in message["result"].keys()
+        or "type" not in message["result"]["result"].keys()
+    ):
+        raise NodeDebuggerUnexpectedResponse(message) from None
+    if expected_result == ResultType.BOOLEAN:
+        if message["result"]["result"]["type"] != "boolean":
+            raise NodeDebuggerUnexpectedResponse(message) from None
+    elif expected_result == ResultType.INTEGER:
+        if message["result"]["result"]["type"] != "number":
+            raise NodeDebuggerUnexpectedResponse(message) from None
+    elif expected_result == ResultType.STRING:
+        if message["result"]["result"]["type"] != "string":
+            raise NodeDebuggerUnexpectedResponse(message) from None
```

Comment:
You can use your enum values :)
```suggestion
    # check if result is not boolean/number/string as expected
    if expected_result.value != message["result"]["result"]["type"]:
        raise NodeDebuggerUnexpectedResponse(message) from None
```

Reply:
I suggested that the entire set of `if`s can be removed, but that can be done in the next PR